src/compiler/program.ts

```diff
@@ -1507,6 +1521,97 @@ namespace ts {
             });
         }
 
+        function findLibraryDefinition(searchPath: string) {
+            let typingFilename = "index.d.ts";
+            const packageJsonPath = combinePaths(searchPath, "package.json");
+            if (host.fileExists(packageJsonPath)) {
+                let package: { typings?: string } = {};
+                try {
+                    package = JSON.parse(host.readFile(packageJsonPath));
+                }
+                catch (e) { }
+
+                if (package.typings) {
+                    typingFilename = package.typings;
+                }
+            }
+
+            const combinedPath = normalizePath(combinePaths(searchPath, typingFilename));
+            return host.fileExists(combinedPath) ? combinedPath : undefined;
+        }
+
+        function processReferencedLibraries(file: SourceFile, compilationRoot: string) {
+            const primarySearchPaths = map(getEffectiveLibraryPrimarySearchPaths(), path => combinePaths(compilationRoot, path));

```

Comment:
I do not see `getEffectiveLibraryPrimarySearchPaths` used else where, so consider pushing combining paths to it as well, since the path coming from the tsconfig.json are already mapped to the root directory, so it is only the default ones that we need to combine them with the root.